syntax = "proto3";

package orchestrator;

service Orchestrator {
  rpc Register(RegisterRequest) returns (RegisterResponse);
  rpc SendHeartbeat(HeartbeatRequest) returns (HeartbeatResponse);
}

message RegisterRequest {
  string worker_id = 1;
}

message RegisterResponse {
  int32 assigned_rank = 1;
  int32 world_size = 2;
  bool success = 3;
  string message = 4;  // Error message if success=false
}

message HeartbeatRequest {
  string worker_id = 1;
  int64 timestamp = 2;
}

message HeartbeatResponse {
  bool acknowledged = 1;
}